### PR TITLE
Add universal chat pattern

### PR DESCRIPTION
**Pull request by wvffle**
To test the regex you can use https://www.regexpal.com/

**Chat regex**
```js
/^(?:\(.+\)|\[.+\]|.)*?(\w+)\s?[>:\-»\]\)~]+\s(.*)$/
```
**Chat tests**
```
Teleported flatbot to 0.5, 4.0, 0.5
[Server: Opped flatbot]
Given  (ID 17) * 255 to flatbot for 1 seconds
Set the display objective in slot 'list' to 'test2'
//-- above should drop --//
wvffle : meh
wvffle: meh
wvffle: meh
wvffle ~ meh
<wvffle> meh
<wvffle> meh (meh)
wvffle: (meh)
moderator wvffle: meh
[wvffle] meh
[a] [b]wvffle » meh
[a] [b] wvffle » meh
[a] ~wvffle: meh
[] wvffle - meh
```

**Whisper regex**
```js
/^(?:\(.+\)|\[.+\]|.)*?(\w+) whispers (?:to you)?:? (.*)$/
/^(?:\(.+\)|\[.+\]|.)*?(\w+) -> \w+\\s?\] (.*)$/ // - \w is there instead of 'me' to support translations
```
**Whisper tests**
```
[wvffle -> me] meh
wvffle whispers meh
wvffle whispers: meh
wvffle whispers to you meh
```
wvffle whispers to you: meh

### CODE REVIEW

**Comment by wvffle** [comment]
All of those passes:
image

**Comment by wvffle** [comment]
image

All above examples pass

**Comment by wvffle** [comment]
What about #175?

**Comment by wvffle** [comment]
The only error right now is that we match some server replies
```
username: Server
message: Opped flatbot]
```
and
```
username: Given
message: (ID 17) * 255 to flatbot for 1 seconds
```
image

File: package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
   },
   "license": "MIT",
   "dependencies": {
-    "minecraft-data": "^2.31.0",
+    "minecraft-data": "^2.36.0",
```
Comment (rom1504) [inline package.json]: Why ?

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "prismarine-item": "^1.0.1",
     "prismarine-recipe": "^1.0.1",
     "prismarine-windows": "^1.1.0",
-    "protodef": "^1.6.7",
```
Comment (rom1504) [inline package.json]: Why ?

1. Unrelated with this review
2. Not needed because of the ^

Comment (wvffle) [reply package.json]: Just merged two of the update branches. I think that dependencies should be updated. Especially minecraft-data ;)

**Comment by wvffle** [comment]
current regex to match `chat` is 
```js
/^(?:\(.+\)|\[.+\]|.)*?(\w+)\s?[>:\-»\]\)~]+\s(.*)$/
```
so basically it matches
image

**Comment by rom1504** [comment]
How many chat plugins does it work on ? Can you add in the doc what chat plugins that add support of ?

**Comment by wvffle** [comment]
I don't really know. I haven't tested it with plugins. I just typed 'minecraft chat' and tried to match all of different chat patterns in screenshots.

The only downside of this approach is that it can match some custom plugin messages.
Messages starting with `[Server:` are filtered out by default.

**Pull request merged**